### PR TITLE
create @types/netgear

### PR DESCRIPTION
**Pull request by k-yle**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/netgear/netgear-tests.ts [hunk 1]
```diff
@@ -0,0 +1,255 @@
+import Netgear from 'netgear';
```
Comment (peterblazejewicz) [inline types/netgear/netgear-tests.ts:1]:
```suggestion
import Netgear = require('netgear');
```
https://github.com/gruijter/netgear.js#quickstart

File: types/netgear/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "module": "commonjs",
+        "lib": ["es6", "dom"],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": ["../"],
+
+        "esModuleInterop": true,
```
Comment (peterblazejewicz) [inline types/netgear/tsconfig.json:12]:
```suggestion
```
you don't need this (with `= reuire(....)`, it's for people using this DT in the wild

**Review by peterblazejewicz — COMMENTED** [review]
Looks OK,I believe you should name instances of class and namespace `NetgearRouter`,
that's OK, the package itself needs to follow NPM package name, but real content can be named just like actual one, so:
```ts
export = NetgearRouter;
```
https://github.com/gruijter/netgear.js/blob/master/netgear.js#L2041

what do you think?

**Comment by k-yle** [comment]
@peterblazejewicz sounds good. CI is failing on master so I'll fix this tomorrow

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@k-yle thx!

**Comment by orta** [comment]
👍🏻

**Pull request merged**